src/main/java/edu/harvard/iq/dataverse/dataverse/messages/DataverseTextMessageServiceBean.java

```diff
@@ -24,11 +25,60 @@
     private static final Logger logger = Logger.getLogger(DataverseTextMessageServiceBean.class.getCanonicalName());
 
     @Inject
-    private DataverseSession session;
+    private DataverseMessagesMapper mapper;
 
     @PersistenceContext(unitName = "VDCNet-ejbPU")
     private EntityManager em;
 
+    DataverseTextMessageServiceBean() {
+    }
+
+    DataverseTextMessageServiceBean(EntityManager em, DataverseMessagesMapper mapper) {
+        this.em = em;
+        this.mapper = mapper;
+    }
+
+    public DataverseTextMessageDto newTextMessage(Long dataverseId) {
```

Comment:
To nie powinno byc w maperze?

Reply:
Nie, bo logika jest tutaj jak stworzyć nowego dto do edycji: czyli new text message albo do edycji "reuse". A mapper mapuje tylko wartosci

Reply:
Moze byc